pkg/provider/azure_utils.go

```diff
@@ -99,26 +99,37 @@ func convertMapToMapPointer(origin map[string]string) map[string]*string {
 	return newly
 }
 
-func parseTags(tags string) map[string]*string {
+func parseTags(tags string, tagsMap map[string]string) map[string]*string {
 	formatted := make(map[string]*string)
-	if len(tags) == 0 {
-		return formatted
-	}
 
-	kvs := strings.Split(tags, consts.TagsDelimiter)
-	for _, kv := range kvs {
-		res := strings.Split(kv, consts.TagKeyValueDelimiter)
-		if len(res) != 2 {
-			klog.Warningf("parseTags: error when parsing key-value pair %s, would ignore this one", kv)
-			continue
+	if tags != "" {
+		kvs := strings.Split(tags, consts.TagsDelimiter)
+		for _, kv := range kvs {
+			res := strings.Split(kv, consts.TagKeyValueDelimiter)
+			if len(res) != 2 {
+				klog.Warningf("parseTags: error when parsing key-value pair %s, would ignore this one", kv)
+				continue
+			}
+			k, v := strings.TrimSpace(res[0]), strings.TrimSpace(res[1])
+			if k == "" || v == "" {
+				klog.Warning("parseTags: empty key or value, ignoring this key-value pair")
+				continue
+			}
+			formatted[k] = to.StringPtr(v)
 		}
-		k, v := strings.TrimSpace(res[0]), strings.TrimSpace(res[1])
-		if k == "" || v == "" {
-			klog.Warningf("parseTags: error when parsing key-value pair %s-%s, would ignore this one", k, v)
-			continue
+	}
+
+	if len(tagsMap) > 0 {
+		for key, value := range tagsMap {
+			key, value := strings.TrimSpace(key), strings.TrimSpace(value)
+			if key == "" || value == "" {
+				klog.Warningf("parseTags: empty key or value, ignoring this key-value pair")
+				continue
+			}
+			formatted[key] = to.StringPtr(value)
```

Comment:
Also check the existing keys in case there're existing ones with a different character size?

Reply:
it's checked in az.reconcileTags

Reply:
If there're keys with different values set in tags and tagsMap, reconcileTags couldn't ensure which one would be used. Hence, it's better to report an error here

Reply:
In this case `tags` and `tagsMap` will be merged into a map[string]*string and the values in `tagsMap` will replace those in `tags` if there are the same keys. This new map will be used in az.reconcileTags and it will not call the update api if there are case conflicts between existing tags on resources and that map.

Reply:
Does it work as expected?